vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```
@@ -48,33 +57,153 @@ public class ElementQuery<T extends TestBenchElement> {
-            if (operator != null) {
-                neg = operator.startsWith("!");
-                oper = neg ? operator.substring(1) : operator;
-                oper = oper.isEmpty() ? null : oper;
+            public boolean isNeg() {
+                return neg;
             }
+        }
+
+        private final String name;
+        private final Operator operator;
+        private final String value;
 
-            this.operator = oper;
+        /**
+         * Instantiates an attribute matching expression
+         * having the supplied attribute name, comparison operator, and
+         * value to compare with the attribute's value.
+         *
+         * @param name the name of the attribute
+         * @param operator the comparison operator to use in the comparison
+         * @param value the value to compare with the attribute's value
+         */
+        public AttributeMatch(String name, Operator operator, String value) {
+            this.name = name;
+            this.operator = operator;
             this.value = value;
-            this.negate = neg;
         }
 
+        /**
+         * Instantiates an attribute matching expression
+         * having the supplied attribute name, comparison operator, and
+         * value to compare with the attribute's value.
+         *
+         * @param name the name of the attribute
+         * @param operator the comparison operator to use in the comparison
+         * @param value the value to compare with the attribute's value
+         *
+         * @deprecated use {@link #AttributeMatch(String, Operator, String)}
+         */
+        @Deprecated(since = "9.3")
+        public AttributeMatch(String name, String operator, String value) {
+            this(name,
+                    Arrays.stream(Operator.values())
+                            .filter(op -> op.getOp().equals(operator))
+                            .findFirst()
+                            .orElse(EXISTS),
```

Previously, we (probably mistakenly) accepted any operator the developer provided; now we turn unknown operators into `EXISTS`.

We should either document the new behavior or throw an exception.

In both cases, this is a breaking change.

---

I prefer the exception option, as invalid operators fail upon use, so should not be allowed, and we don't want them failing silently.